src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1182,6 +1182,16 @@ FixedwingPositionControl::control_auto_position(const hrt_abstime &now, const fl
 	if (_param_fw_use_npfg.get()) {
 		_npfg.setAirspeedNom(target_airspeed * _eas2tas);
 		_npfg.setAirspeedMax(_param_fw_airspd_max.get() * _eas2tas);
+
+		if (_control_mode.flag_control_offboard_enabled && PX4_ISFINITE(pos_sp_curr.vx) && PX4_ISFINITE(pos_sp_curr.vy)) {
+			// Navigate directly on position setpoint and path tangent
+			matrix::Vector2f velocity_2d(pos_sp_curr.vx, pos_sp_curr.vy);
+			_npfg.navigatePathTangent(curr_pos, curr_wp, velocity_2d.normalized(), ground_speed, _wind_vel, pos_sp_curr.yawspeed);
```

Comment:
maybe good to make a comment here noting that yawspeed is hijacked for path curvature - not sure if there should be some additional documentation in the position setpoint msg about this alternate offboard use case.. or if we leave it for now until a cleaner path interface can be made

Reply:
Ah, I forgot about the hijacking I did. Should we try and use the acceleration setpoint? I can also remove it from the PR for now

Reply:
This has been removed now. Curvature is passed as acceleration setpoints